python/ray/autoscaler/_private/command_runner.py

```diff
@@ -318,7 +318,7 @@ def run(
         if shutdown_after_run:
             cmd += "; sudo shutdown -h now"
         if ssh_options_override_ssh_key:
-            ssh_options = SSHOptions(ssh_options_override_ssh_key)
+            ssh_options = SSHOptions(ssh_options_override_ssh_key, ProxyCommand=self.ssh_proxy_command)
```

Comment:
I'm not sure what the consequence would be if self.ssh_proxy_command is the default `None`.
To de-risk this, would you mind writing this as follows: 
```suggestion
            if self.ssh_proxy_command:
                ssh_options = SSHOptions(ssh_options_override_ssh_key, ProxyCommand=self.ssh_proxy_command)
            else:
                ssh_options = SSHOptions(ssh_options_override_ssh_key)
```

Reply:
I don't know a lot about this code path.

The git blame on this points to a PR of @AmeerHajAli reviewed by @ericl and @ijrsvt in August 2020.
@AmeerHajAli @ericl @ijrsvt any thoughts?

Reply:
I think it ends up passed directly to SSH as ProxyCommand=None, which is probably bad. We should avoid setting it if None.

Reply:
Thanks @DmitriGekhtman and @ericl I've updated the PR. It looks like Proxycommand does not get passed to ssh if it is None, see: https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/_private/command_runner.py#L188-L192, still, to be careful, I updated the PR like you suggested.